### PR TITLE
Traffic weighting

### PR DESCRIPTION
**Pull request by mark-church**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This adds a documentation guide with realistic examples for traffic shifting.

Question for reviewers: There are many words for this - traffic shifting, traffic splitting, traffic weighting. Is there a preference that we have in this project?

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/service-apis/issues/510

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by robscott** [comment]
Thanks!

/ok-to-test

File: docs-src/traffic-shifting.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# HTTP traffic shifting
```
Comment (robscott) [inline docs-src/traffic-shifting.md:1]: Tiny Nit: Is "Traffic Splitting" or "Traffic Shifting" a more common term? It looks like both terms are used throughout the doc, just trying to think of which users will be more likely to search for.

File: docs-src/traffic-shifting.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+spec:
+  rules:
+  - forwardTo:
+    - serviceName: foo-svc-1
+      port: 8080
+      weight: 90
+    - serviceName: foo-svc-2
+      port: 8080
+      weight: 10
+...
```
Comment (robscott) [inline docs-src/traffic-shifting.md:28]: What do the dots here represent? I'm probably forgetting something obvious, but would this be a valid route as is?

File: docs-src/traffic-shifting.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+This example assumes that the following Gateway is deployed:
+
+```yaml 
+{% include 'traffic-shifting/gateway.yaml' %}   
+```
+
+## Canary traffic
+
+At first, there may only be a single version of a Service that serves
+production user  traffic for `foo.example.com`. The following HTTPRoute has no
```
Comment (robscott) [inline docs-src/traffic-shifting.md:52]: Tiny nit:
```suggestion
production user traffic for `foo.example.com`. The following HTTPRoute has no
```

File: examples/traffic-shifting/foo-traffic-shift-1.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    - serviceName: foo-svc-v1
+      port: 8080
+  - matches:
+    - headers:
+        type: Exact
+        values:
+          traffic: test
+    forwardTo:
+    - serviceName: foo-svc-v2
+      port: 8080
```
Comment (robscott) [inline examples/traffic-shifting/foo-traffic-shift-1.yaml:21]: Nit: need a new line

File: examples/traffic-shifting/foo-traffic-shift-2.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  hostnames:
+  - foo.example.com
+  rules:
+  - forwardTo:
+    - serviceName: foo-svc-v1
+      port: 8080
+      weight: 90
+    - serviceName: foo-svc-v2
+      port: 8080
+      weight: 10
```
Comment (robscott) [inline examples/traffic-shifting/foo-traffic-shift-2.yaml:17]: Nit: need a new line

File: examples/traffic-shifting/foo-traffic-shift-3.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  hostnames:
+  - foo.example.com
+  rules:
+  - forwardTo:
+    - serviceName: foo-svc-v1
+      port: 8080
+      weight: 0
+    - serviceName: foo-svc-v2
+      port: 8080
+      weight: 1
```
Comment (robscott) [inline examples/traffic-shifting/foo-traffic-shift-3.yaml:17]: Nit: need a new line

File: examples/traffic-shifting/gateway.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - protocol: HTTP
+    port: 80
+    routes:
+      kind: HTTPRoute
+      selector:
+        matchLabels:
+          gateway: prod-web-gw
```
Comment (robscott) [inline examples/traffic-shifting/gateway.yaml:14]: Nit: need a new line

**Review by robscott — COMMENTED** [review]
Thanks @mark-church! This will be a great addition to our docs! Mostly LGTM, just had a few nits

File: docs-src/traffic-shifting.md (hunk 1 above)
Comment (robscott) [reply docs-src/traffic-shifting.md:1]: Just saw that you asked about this in the PR as well. Between traffic shifting, traffic splitting, and traffic weighting, I have a slight preference towards traffic splitting, followed by traffic shifting. Interested in what others think as well.

Comment (hbagdi) [reply docs-src/traffic-shifting.md:1]: I like traffic splitting over shifting.

File: docs-src/traffic-shifting.md (hunk 2 above)
Comment (hbagdi) [reply docs-src/traffic-shifting.md:28]: I think it is best to `include` a valid HTTPRoute resource here. It is verbose but allows users to copy-paste and hit the ground running.

**Comment by hbagdi** [comment]
Other than comment from Rob, this looks nearly ready for merge.
@mark-church Thanks for putting this together, this is good stuff!

File: docs-src/traffic-shifting.md (hunk 2 above)
Comment (mark-church) [reply docs-src/traffic-shifting.md:28]: okay I added in the full resource with an include

File: docs-src/traffic-shifting.md (hunk 1 above)
Comment (mark-church) [reply docs-src/traffic-shifting.md:1]: traffic splitting wins then! I'm switching over everything for traffic splitting.

**Comment by mark-church** [comment]
Latest changes ....

- Changed everything from "traffic shifting" to "traffic splitting"
- Use only includes from examples folder
- Updated and simplified examples

**Comment by robscott** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**